crates/pep508-rs/src/lib.rs

```diff
@@ -882,6 +1019,170 @@ fn preprocess_url(
     }
 }
 
+/// Like [`parse_url`], but allows for extras to be present at the end of the URL, to comply
+/// with the non-PEP 508 extensions.
+///
+/// For example:
+/// - `https://download.pytorch.org/whl/torch_stable.html[dev]`
+/// - `../editable[dev]`
+fn parse_unnamed_url(
+    cursor: &mut Cursor,
+    working_dir: Option<&Path>,
+) -> Result<(VerbatimUrl, Vec<ExtraName>), Pep508Error> {
+    // wsp*
+    cursor.eat_whitespace();
+    // <URI_reference>
+    let (start, len) = cursor.take_while(|char| !char.is_whitespace());
+    let url = cursor.slice(start, len);
+    if url.is_empty() {
+        return Err(Pep508Error {
+            message: Pep508ErrorSource::String("Expected URL".to_string()),
+            start,
+            len,
+            input: cursor.to_string(),
+        });
+    }
+
+    let url = preprocess_unnamed_url(url, working_dir, cursor, start, len)?;
+
+    Ok(url)
+}
+
+/// Create a `VerbatimUrl` to represent the requirement, and extracts any extras at the end of the
+/// URL, to comply with the non-PEP 508 extensions.
+fn preprocess_unnamed_url(
+    url: &str,
+    #[cfg_attr(not(feature = "non-pep508-extensions"), allow(unused))] working_dir: Option<&Path>,
+    cursor: &Cursor,
+    start: usize,
+    len: usize,
+) -> Result<(VerbatimUrl, Vec<ExtraName>), Pep508Error> {
+    // Split extras _before_ expanding the URL. We assume that the extras are not environment
+    // variables.
```

Comment:
Just because we don't want to support this or is it hard for some reason?

Reply:
Both... I'll expand in a comment here, it's a good question.